### PR TITLE
upgrade temporal to v1.16.1

### LINKED ISSUE #46: Does not build on Windows
## Expected Behavior

Builds on Windows.

## Actual Behavior

Not building. From a fresh recursive git clone, a `go build ./cmd/temporalite` gives:

    C:\Users\user\go\pkg\mod\github.com\temporalio\ui-server@v0.1.1-0.20211223210854-4484839e0398\server\server.go:47:113: pattern generated/ui/_app/assets/pages/namespaces/\[namespace\]/queues/\[queue\]/*: syntax error in pattern

Since the UI server was added, https://github.com/temporalio/ui-server/issues/104 is causing problems with Windows build. Supporting the UI as a build flag (or one prefixed with a bang for opt-out) per #33 could alleviate this temporarily. Also, I'd recommend adding Windows do the CI workflow.

This is on Go 1.17.7. I have not investigated the fix, this is just for tracking.

### PR DESCRIPTION
**Pull request by balchua**
**What changed?**
Upgrade to temporal v1.16.1, recently released and `ui-server` to `v0.13.0`.  This also fixes #46

**Why?**
To keep pace with temporal releases

**How did you test it?**
Tested locally, by creating workflows.

**Potential risks**
There is a breaking API reported in temporal.  Refer to [this](https://github.com/temporalio/sdk-go/pull/706)

**Is hotfix candidate?**
No

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -133,10 +133,7 @@ func (s *Server) NewClient(ctx context.Context, namespace string) (client.Client
 // Note that the HostPort and ConnectionOptions fields of client.Options will always be overridden.
 func (s *Server) NewClientWithOptions(ctx context.Context, options client.Options) (client.Client, error) {
 	options.HostPort = s.frontendHostPort
-	options.ConnectionOptions = client.ConnectionOptions{
-		DisableHealthCheck: false,
-		HealthCheckTimeout: timeoutFromContext(ctx, time.Minute),
-	}
+	options.ConnectionOptions = client.ConnectionOptions{}
```
Comment (cretz) [inline server.go:136]: FWIW, this isn't necessary (it'll be an empty struct anyways)

Comment (balchua) [reply server.go]: thanks for pointing this out.  just removed the unused connection options.

**Comment by jlegrone** [comment]
Hey @balchua, do you mind updating https://github.com/DataDog/temporalite/blob/main/.github/workflows/release.yml and https://github.com/DataDog/temporalite/blob/main/.github/workflows/ci.yml to Go 1.18? I think that should fix the build.

**Review by jlegrone — APPROVED** [review]

**Comment by balchua** [comment]
> Hey @balchua, do you mind updating https://github.com/DataDog/temporalite/blob/main/.github/workflows/release.yml and https://github.com/DataDog/temporalite/blob/main/.github/workflows/ci.yml to Go 1.18? I think that should fix the build.

Sure.  I will bump up the `ui-server` dependency to the `v0.12.2`.  I also tried to build it on windows and it builds successfully.

**Comment by jlegrone** [comment]
Thanks @balchua! I ended up needing to push a couple more changes to update docs and test for #46, so opened #66 based on your changes. Closing this PR now that #66 is merged.

**Pull request closed without merging**